temporalcli/commandsmd/commands.md

```diff
@@ -209,6 +210,107 @@ Cluster commands follow this syntax: `temporal operator cluster [command] [comma
 * `--frontend-address` (string) - IP address to bind the frontend service to. Required.
 * `--enable-connection` (bool) - enable cross cluster connection.
 
+### temporal operator namespace: Operations performed on Namespaces.
+
+Namespace commands perform operations on Namespaces contained in the Temporal Cluster.
+
+Cluster commands follow this syntax: `temporal operator namespace [command] [command options]`
+
+### temporal operator namespace create [namespace]: Registers a new Namespace.
+
+The temporal operator namespace create command creates a new Namespace on the Server.
+Namespaces can be created on the active Cluster, or any named Cluster.
+`temporal operator namespace create --cluster=MyCluster example-1`
+
+Global Namespaces can also be created.
+`temporal operator namespace create --global example-2`
+
+Other settings, such as retention and Visibility Archival State, can be configured as needed.
+For example, the Visibility Archive can be set on a separate URI.
+`temporal operator namespace create --retention=5 --visibility-archival-state=enabled --visibility-uri=some-uri example-3`
+
+<!--
+* exact-args=1
+-->
+
+#### Options
+
+* `--active-cluster` (string) - Active cluster name.
+* `--cluster` (string[]) - Cluster names.
+* `--data` (string) - Namespace data in key=value format. Use JSON for values.
+* `--description` (string) - Namespace description.
+* `--email` (string) - Owner email.
+* `--global` (bool) - Whether the namespace is a global namespace.
+* `--history-archival-state` (string-enum) - History archival state. Options: disabled, enabled. Default: disabled.
+* `--history-uri` (string) - Optionally specify history archival URI (cannot be changed after first time archival is enabled).
+* `--retention` (string) - Length of time (in days) a closed Workflow is preserved before deletion. Default: 3.
+* `--visibility-archival-state` (string-enum) - Visibility archival state. Options: disabled, enabled. Default: disabled.
+* `--visibility-uri` (string) - Optionally specify visibility archival URI (cannot be changed after first time archival is enabled).
+
+### temporal operator namespace delete [namespace]: Deletes an existing Namespace.
+
+The temporal operator namespace delete command deletes a given Namespace from the system.
+
+<!--
+* exact-args=1
+-->
+
+#### Options
+
+* `--yes`, `-y` (bool) - Confirm prompt to perform deletion.
+
+### temporal operator namespace describe [namespace]: Describe a Namespace by its name or ID.
+
+The temporal operator namespace describe command provides Namespace information.
+Namespaces are identified by Namespace ID.
+
+`temporal operator namespace describe --namespace-id=some-namespace-id`
+`temporal operator namespace describe example-namespace-name`
+
+<!--
+* maximum-args=1
```

Comment:
A flag for namespace name would have been better.

But we don't use the global --namespace flag in this command as it has a different meaning. In that case, we will need a new flag like --namespace-name.

Also, the previous cli will fail if `--namespace` is set when using namespace commands. But I could not add that check here because we set it to default namespace.

Reply:
I agree, will discuss with team about flag vs option here.